pageserver/src/tenant/timeline/compaction.rs

```diff
@@ -2007,28 +2073,36 @@ impl Timeline {
         let last_key = last_key.expect("no keys produced during compaction");
         // TODO: move this part to the loop body
         stat.on_unique_key_visited();
-        let retention = self
-            .generate_key_retention(
-                last_key,
-                &accumulated_values,
-                gc_cutoff,
-                &retain_lsns_below_horizon,
-                COMPACTION_DELTA_THRESHOLD,
-                get_ancestor_image(self, last_key, ctx).await?,
-            )
-            .await?;
-        // Put the image into the image layer. Currently we have a single big layer for the compaction.
-        retention
-            .pipe_to(
-                last_key,
-                self,
-                &mut delta_layer_writer,
-                image_layer_writer.as_mut(),
-                &mut stat,
-                dry_run,
-                ctx,
-            )
-            .await?;
+
+        let skip_adding_key = if let Some(ref compaction_key_range) = compaction_key_range {
```

Comment:
avoid blank line to make it clear what the TODO refers to.


```suggestion
        let last_key = last_key.expect("no keys produced during compaction");
        // TODO: move this part to the loop body
        stat.on_unique_key_visited();
        let skip_adding_key = if let Some(ref compaction_key_range) = compaction_key_range {
```